### PR TITLE
Send Message to Direct Chat Mutation

### LINKED ISSUE #1313: Send message to a direct chat
**Is your feature request related to a problem? Please describe.**
A user must be able to type their message in the text field and click the send button.

**Describe the solution you'd like**
Use SendMessageToDirectChat mutation to send a message to a specific direct chat.

**Describe alternatives you've considered**
N.A.

**Approach to be followed (optional)**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by SriramPatibanda**
**What kind of change does this PR introduce?**

- Adds send message to direct chat feature
- Queries for subscription
- Work In Progress: Using Subscription for instant reflection of message

Fixes #1313 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

N.A.

**If relevant, did you update the documentation?**

N.A

**Summary**

Now, one can send a message by typing it in the text field of direct chat. Though the message would be sent to the direct chat, it won't reflect directly on the UI as messages are being fetched using graphql query as of now. GraphQL subscriptions are WIP using which messages will be reflected instantly in future.

**Does this PR introduce a breaking change?**

No

**Other information**
N.A>

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please create an issue for this PR and update the PR comments so that it autocloses the issue when the PR is merged.

**Comment by palisadoes** [comment]
@DeltaHarbinger @TheHazeEffect Please review. This is an important feature that is being added.

**Comment by SriramPatibanda** [comment]
Can someone please review this?

File: lib/services/graphql_config.dart [hunk 1]
```diff
@@ -32,10 +32,19 @@ class GraphqlConfig {
     authClient();
   }
 
+  static final WebSocketLink _webSocketLink = WebSocketLink(
+    'ws://sriram.dscnitrourkela.org/graphql',
```
Comment (DeltaHarbinger) [inline lib/services/graphql_config.dart:36]: Hey, was this link a test link? Should it be replaced before merging?

File: lib/services/chat_service.dart [hunk 2]
```diff
@@ -28,6 +31,30 @@ class ChatService {
...
+        ChatQueries().sendMessageToDirectChat(),
+        variables: {"chatId": chatId, "messageContent": messageContent});
+
+    final message = ChatMessage.fromJson(
+      result.data['sendMessageToDirectChat'] as Map<String, dynamic>,
+    );
+
+    _chatMessageController.add(message);
+
+    print(result.data);
```
Comment (DeltaHarbinger) [inline lib/services/chat_service.dart:55]: Do you want to remove print statements before merging?

File: lib/services/chat_service.dart [hunk 3]
```diff
@@ -28,6 +31,30 @@ class ChatService {
   Stream<ChatListTileDataModel> get chatListStream => _chatListStream;
   Stream<ChatMessage> get chatMessagesStream => _chatMessagesStream;
 
+  // Stream<QueryResult> getMessagesFromDirectChat() async* {
```
Comment (DeltaHarbinger) [inline lib/services/chat_service.dart:34]: Should comments be removed before merging?

Comment (SriramPatibanda) [reply lib/services/chat_service.dart:34]: I want to let this stay. I'll be using this code for subscriptions.

File: lib/services/graphql_config.dart (hunk 1 above)
Comment (SriramPatibanda) [reply lib/services/graphql_config.dart:36]: Yeah, this was a test link. I'll make the needed changes.

**Comment by palisadoes** [comment]
@SriramPatibanda please fix the failing tests

**Comment by SriramPatibanda** [comment]
Please go ahead and merge the PR so that @nishnatadebnath can work on the final part of direct chats

**Review by palisadoes — APPROVED** [review]

**Pull request merged**